### PR TITLE
CVE-2015-1237 and CVE-2011-2858

### PR DESCRIPTION
**Pull request by MEF4232**
Updated CVE yml files

### CODE REVIEW

File: cves/CVE-2011-2858.yml [hunk 1]
```diff
@@ -59,20 +59,33 @@ fixes_vcc_instructions: |
...
 - :commit: 1e8187453cbfeab13f1d229c91b66b2b2201a578
   :note: ''
-vccs: []
+vccs: [
+- :commit: fe0433b999d2bc6a4b717d546ebc0108f35eed34
+  :note: Fix bug in SimulateAttrib0 that did not check for out of memory. It also 
+  did not correctly check for math overflow. Also fixed similar bugs in 
+  SimulateFixedAttribs
+  :date of commit: Sep 9, 2011
+]
```
Comment (Jeng22) [inline cves/CVE-2011-2858.yml]: Dates should be in YYYY-MM-DD format. Similar to the announced field.

File: cves/CVE-2011-2858.yml [hunk 2]
```diff
@@ -147,8 +160,26 @@ major_events:
...
+- :commit: 3c7303a2fa6b76aa415e3e51241288f95eb80923
+  :note: (1) Revert 100384 - Fix bug in SimulateAttrib0 that did not check for out 
+  of memory.
+  :team member: nsylvain@chromium.org
+- :commit: 9b4ed138d5bd09082eb63bc4ea8681014676579b
+  :note: ''
+- :commit: 1e8187453cbfeab13f1d229c91b66b2b2201a578
+  :note: (2) Revert 100453 - Revert "Revert 100384 - Fix bug in SimulateAttrib0 that 
+  did not check for out of memory."
+  :team member: dpapad@chromium.org
```
Comment (Jeng22) [inline cves/CVE-2011-2858.yml]: I like the addition of the team member associated with the commits.

File: cves/CVE-2015-1237.yml [hunk 3]
```diff
@@ -76,7 +81,7 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
+  answer: TEST=No more crashes in RenderFrameImpl::OnMessageReceived
```
Comment (Jeng22) [inline cves/CVE-2015-1237.yml]: What did the test do? what is a modified test or creation of a new one?

File: cves/CVE-2011-2858.yml [hunk 4]
```diff
@@ -11,7 +11,7 @@ curated_instructions: |
   integrity checks on this file to make sure you fill everything out properly.
```
Comment (kayladavis) [inline cves/CVE-2011-2858.yml]: Please fill out the "description" section with a quick summary of what happened.

File: cves/CVE-2011-2858.yml (hunk 2 above)
Comment (MEF4232) [reply cves/CVE-2011-2858.yml]: Thanks!

File: cves/CVE-2015-1237.yml [hunk 5]
```diff
@@ -32,7 +32,13 @@ description_instructions: |
...
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Use-after-free means referencing memory after the location has been 
+freed. This use-after-free vulnerability allows a remote attacker to cause 
+a denial of service (one of the STRIDE security threats). It can also trigger 
+renderer IPC messages during a detach operation. The "ipcRenderer" module provides 
+the ability to send and receive synchronous and asynchronous messages from the 
+renderer process to the main process. A detach operation involves dependencies; 
+if the parent entity is detached, the related entity will also be detached.
```
Comment (Jeng22) [inline cves/CVE-2015-1237.yml]: So what is being used after freed? It sounds like the IPC rendering  messages is an additional effect of the problem. 
I like the explanation of each term/module.

File: cves/CVE-2011-2858.yml [hunk 6]
```diff
@@ -84,7 +87,7 @@ vccs: [
   :note: Fix bug in SimulateAttrib0 that did not check for out of memory. It also 
   did not correctly check for math overflow. Also fixed similar bugs in 
   SimulateFixedAttribs
-  :date of commit: Sep 9, 2011
+  :date of commit: 2011-09-09
 ]
 upvotes_instructions: |
   For the first round, ignore this upvotes number.
```
Comment (Jeng22) [inline cves/CVE-2011-2858.yml]: +3 Upvotes for this

File: cves/CVE-2015-1237.yml (hunk 3 above)
Comment (MEF4232) [reply cves/CVE-2015-1237.yml]: I updated this answer

File: cves/CVE-2011-2858.yml [hunk 7]
```diff
@@ -32,7 +32,10 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Triangle arrays are not properly handled which allows remote attackers
```
[the remark this answers is not in the document]
Comment (MEF4232) [reply cves/CVE-2011-2858.yml]: I added to the description. Here is a great visual https://en.wikipedia.org/wiki/Triangular_array

File: cves/CVE-2015-1237.yml (hunk 5 above)
Comment (MEF4232) [reply cves/CVE-2015-1237.yml]: Thanks! Updated now

**Comment by MEF4232** [comment]
@tanner-glantz I made some changes but my checks are still "failing" due to your request. Please take a look at the updates. Thanks

File: cves/CVE-2011-2858.yml [hunk 8]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: CWE-119: Buffer Errors
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: The colon after CWE-119 should be removed to fix the yaml formal error.

File: cves/CVE-2011-2858.yml [hunk 9]
```diff
@@ -32,7 +32,11 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Triangle arrays (each row is length equals the index of the row)
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: why are triangle arrays used?

File: cves/CVE-2011-2858.yml [hunk 10]
```diff
@@ -59,20 +63,33 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 9b4ed138d5bd09082eb63bc4ea8681014676579b
-  :note: ''
+  :note: (3) Revert 100494 - Revert "Revert 100453 - Revert "Revert 100384 - Fix bug
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: Why was it reverted

Comment (nxc5929) [reply cves/CVE-2011-2858.yml]: The note should be more of what happened

File: cves/CVE-2011-2858.yml [hunk 11]
```diff
@@ -59,20 +63,33 @@ fixes_vcc_instructions: |
   CVE-2011-3092.yml). Fixes and VCCs follow the same format.
 fixes:
 - :commit: 9b4ed138d5bd09082eb63bc4ea8681014676579b
-  :note: ''
+  :note: (3) Revert 100494 - Revert "Revert 100453 - Revert "Revert 100384 - Fix bug 
+  in SimulateAttrib0.""
+  :date of commit: 2011-09-09
 - :commit: 3c7303a2fa6b76aa415e3e51241288f95eb80923
-  :note: ''
+  :note: (1) Revert 100384 - Fix bug in SimulateAttrib0 that did not check for out
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: I would remove "Revert 100384"

File: cves/CVE-2011-2858.yml [hunk 12]
```diff
@@ -59,20 +63,33 @@ fixes_vcc_instructions: |
...
+  in SimulateAttrib0.""
+  :date of commit: 2011-09-09
 - :commit: 3c7303a2fa6b76aa415e3e51241288f95eb80923
-  :note: ''
+  :note: (1) Revert 100384 - Fix bug in SimulateAttrib0 that did not check for out 
+  of memory.
+  :date of commit: 2011-09-09
 - :commit: c13e1da62b5f5f0e6fe8c1f769a5a28415415244
-  :note: ''
+  :note: (4) Revert "Revert 100494 - Fix bug in SimulateAttrib0."
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: I would remove "Revert 100494"

File: cves/CVE-2011-2858.yml [hunk 13]
```diff
@@ -93,9 +110,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: 'TEST=unit tests and run on linux touch (where bug was)'
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: were*

File: cves/CVE-2011-2858.yml [hunk 14]
```diff
@@ -111,10 +130,10 @@ discovered:
 
     If there is no evidence as to how this vulnerability was found, then you may
     leave this part blank.
-  answer: 
-  date: 
+  answer:A bug was discovered that caused a unsigned to signed conversion issue.
```
Comment (nxc5929) [inline cves/CVE-2011-2858.yml]: Questions is focusing more on how is what discovered not so much what the bug was

File: cves/CVE-2015-1237.yml [hunk 15]
```diff
@@ -4,14 +4,14 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 
+CWE: CWE-416: Use After Free
```
Comment (nxc5929) [inline cves/CVE-2015-1237.yml]: The colon after CWE-416 should be removed to fix the yaml formal error.

File: cves/CVE-2015-1237.yml [hunk 16]
```diff
@@ -32,7 +32,13 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: Use-after-free means referencing memory after the location has been 
+freed. This use-after-free vulnerability of a pointer allows a remote attacker to cause 
+a denial of service (one of the STRIDE security threats). It can also trigger 
+renderer IPC messages during a detach operation. The "ipcRenderer" module provides
```
Comment (nxc5929) [inline cves/CVE-2015-1237.yml]: What are IPC messages?

Comment (nxc5929) [reply cves/CVE-2015-1237.yml]: Never midn - It's explained below

**Pull request closed without merging**